mmlearn/tasks/linear_probing.py

```diff
@@ -0,0 +1,454 @@
+        `scheduler` (required) and `extras` (optional). The `scheduler` field should
+        contain configurations for the learning rate scheduler and must include the
+        `_target_` field, which, like the optimizer, should point to the scheduler
+        class. The `extras` field may contain one or more of the following:
+        - `interval` (str): The interval to apply the learning rate scheduler.
+           One of "epoch" or "step". Default is "epoch".
+        - `frequency` (int): The frequency to apply the learning rate scheduler
+          in the specified interval. Default is 1.
+        - `monitor` (str): The metric to monitor for schedulers like ReduceLROnPlateau.
+        - `strict` (bool): Whether to strictly enforce the availability of the
+          monitored metric (if `True`) or raise a warning if the metric is not found
+          (if `False`). Default is `True`.
+
+    Attributes
+    ----------
+    accuracy_metrics : torchmetrics.MetricCollection
+        A collection of metrics that includes accuracy for each `k` in `top_k_list`,
+        providing a comprehensive evaluation of model performance across different
+        levels of top-k predictions.
+    linear_eval : torch.nn.Linear
+        Linear classification layer atop the encoder. Input and output features are
+        determined by `encoder_output_features` and `num_classes`, respectively.
+
+    References
+    ----------
+    [1] He, K., Chen, X., Xie, S., Li, Y., Doll'ar, P., & Girshick, R.B. (2021).
+        Masked Autoencoders Are Scalable Vision Learners. 2022 IEEE/CVF Conference
+        on Computer Vision and Pattern Recognition (CVPR), 15979-15988.
+    """
+
+    def __init__(
+        self,
+        # encoder: torch.nn.Module,
+        encoder: nn.Module,
+        model_checkpoint_path: Optional[str],  # change name
+        modality: str,
+        num_output_features: int,
+        num_classes: int,
+        hidden_dims: Optional[List[int]] = None,
+        task: Literal["binary", "multiclass", "multilabel"] = "multiclass",
+        freeze_encoder: bool = True,
+        pre_classifier_batch_norm: bool = False,
+        top_k_list: Optional[List[int]] = None,
+        optimizer: Optional[partial[torch.optim.Optimizer]] = None,
+        lr_scheduler: Optional[
+            Union[
+                Dict[str, partial[torch.optim.lr_scheduler.LRScheduler]],
+                partial[torch.optim.lr_scheduler.LRScheduler],
+            ]
+        ] = None,
+    ):
+        super().__init__()
+        assert task in ["binary", "multiclass", "multilabel"], (
+            f"Invalid task type: {task}. "
+            "Expected one of ['binary', 'multiclass', 'multilabel']."
+        )
+
+        self.modality = modality
+
+        self.encoder: nn.Module = extract_vision_encoder(
+            encoder, model_checkpoint_path, keys_to_rename={"encoders.rgb.": ""}
+        )
+
+        linear_layer = MLP(num_output_features, num_classes, hidden_dims)
```

Comment:
Expose the rest of the parameters for constructing an MLP (e.g. adding activation functions and drop out layers when using multiple linear layers. This may negate the need for the `pre_classifier_batch_norm` argument.